packages/store/addon/-private/system/core-store.ts

```diff
@@ -341,6 +361,140 @@ class Store extends Service {
     return identifierCacheFor(this);
   }
 
+  _instantiateRecord(
+    internalModel: InternalModel,
+    modelName: string,
+    recordData: RecordData,
+    identifier: StableRecordIdentifier,
+    properties?: { [key: string]: any }
+  ) {
+    if (CUSTOM_MODEL_CLASS) {
+      // assert here
+      if (properties !== undefined) {
+        assert(
+          `You passed '${properties}' as properties for record creation instead of an object.`,
+          typeof properties === 'object' && properties !== null
+        );
+
+        if ('id' in properties) {
+          internalModel.setId(properties.id);
+        }
+
+        // convert relationship Records to RecordDatas before passing to RecordData
+        let defs = this._relationshipsDefinitionFor(modelName);
+
+        if (defs !== null) {
+          let keys = Object.keys(properties);
+          let relationshipValue;
+
+          for (let i = 0; i < keys.length; i++) {
+            let prop = keys[i];
+            let def = defs[prop];
+
+            if (def !== undefined) {
+              if (def.kind === 'hasMany') {
+                if (DEBUG) {
+                  assertRecordsPassedToHasMany(properties[prop]);
+                }
+                relationshipValue = extractRecordDatasFromRecords(properties[prop]);
+              } else {
+                relationshipValue = extractRecordDataFromRecord(properties[prop]);
+              }
+
+              properties[prop] = relationshipValue;
+            }
+          }
+        }
+      }
+
+      // TODO guard against initRecordOptions no being there
+      let createOptions = recordData._initRecordCreateOptions(properties);
+      //TODO Igor pass a wrapper instead of RD
+      let record = this.instantiateRecord(
+        identifier,
+        createOptions,
+        this._recordDataForIdentifier,
+        this._notificationManager
+      );
+      if (IDENTIFIERS) {
+        setRecordIdentifier(record, identifier);
+      }
+      //recordToInternalModelMap.set(record, internalModel);
+      return record;
+    } else {
+      throw 'should not be here, custom model class ff error';
+    }
+  }
+
+  abstract instantiateRecord(
+    identifier: StableRecordIdentifier,
+    createRecordArgs: { [key: string]: any }, // args passed in to store.createRecord() and processed by recordData to be set on creation
+    recordDataFor: (identifier: StableRecordIdentifier) => RecordDataRecordWrapper,
```

Comment:
the signature here should be `RecordIdentifier` not `StableRecordIdentifier` as this is a userland called API